vpr/src/route/parallel_connection_router.cpp

```diff
@@ -903,7 +881,7 @@ void ParallelConnectionRouter::add_route_tree_node_to_heap(
                        describe_rr_node(device_ctx.rr_graph, device_ctx.grid, device_ctx.rr_indexed_data, inode, is_flat_).c_str());
 
 
-        if (tot_cost >= rr_node_route_inf_[inode].path_cost)
+        if (prune_node(inode, tot_cost, backward_path_cost, RREdgeId::INVALID(), target_node, rr_node_route_inf_, cost_params))
```

Comment:
Is pruning node necessary here? Since the start node doesn't need to be pruned inherently.

Reply:
I do agree that this check is a bit unnecessary; however, Master also performs this check. I am not sure why they originally did this (maybe it was a mistake), but I think its safer to leave it just in case it has a purpose. What do you think?

Reply:
Totally agree that we can leave it in to be safe and remove it during the cleanup period.